### PR TITLE
Update 1.x tests infrastructure/libs

### LINKED ISSUE #5630: Update 1.x tests infrastructure / libs
- [ ] Use ganache instead of ganache-cli in 1.x tests
- [ ] Use mocha 10

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

closes #5630 

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -129,7 +129,6 @@
         "lerna": "^4.0.0",
         "mocha": "^6.2.3",
```
Comment (jdevcs) [inline package.json]: will mocha 10 update be done in this PR ?

Comment (nikoulai) [reply package.json]: it's updated now!

**Comment by nikoulai** [comment]
After migrating from `ganache-cli` to `ganache` there are some tests which are failing with no obvious reason. They are skipped for now. After merge, a new issue will be created for investigation and fix.

**Comment by jdevcs** [comment]
> After migrating from `ganache-cli` to `ganache` there are some tests which are failing with no obvious reason. They are skipped for now. After merge, a new issue will be created for investigation and fix.

does these tests relate with this race condition: https://github.com/web3/web3.js/issues/4618#issuecomment-1020422560 ?

**Comment by luu-alex** [comment]
Could you post the errors you run into when you test with new ganache?

**Comment by nikoulai** [comment]
> Could you post the errors you run into when you test with new ganache?

You can see the actions here:

https://github.com/web3/web3.js/actions/runs/3639498168/jobs/6142953341

https://github.com/web3/web3.js/actions/runs/3639825484/jobs/6143699108

https://github.com/web3/web3.js/actions/runs/3638798310/jobs/6141426423

**Comment by jdevcs** [comment]
> > Could you post the errors you run into when you test with new ganache?
> 
> You can see the actions here:
> 
> https://github.com/web3/web3.js/actions/runs/3639498168/jobs/6142953341
> 
> https://github.com/web3/web3.js/actions/runs/3639825484/jobs/6143699108
> 
> https://github.com/web3/web3.js/actions/runs/3638798310/jobs/6141426423

We can ignore eth2 tests but [this](https://github.com/web3/web3.js/actions/runs/3639498168/jobs/6142953341) should not be persistent and should be fixed

**Comment by jdevcs** [comment]
> > > Could you post the errors you run into when you test with new ganache?
> > 
> > 
> > You can see the actions here:
> > https://github.com/web3/web3.js/actions/runs/3639498168/jobs/6142953341
> > https://github.com/web3/web3.js/actions/runs/3639825484/jobs/6143699108
> > https://github.com/web3/web3.js/actions/runs/3638798310/jobs/6141426423
> 
> We can ignore eth2 tests but [this](https://github.com/web3/web3.js/actions/runs/3639498168/jobs/6142953341) should not be persistent and should be fixed

As discussed If these tests are randomly failing due to timeout only , just re run tests in this PR until all are passing and we can merge it.

File: test/e2e.contract.events.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ var Parent = require('./sources/Parent');
 var utils = require('./helpers/test.utils');
 var Web3 = utils.getWeb3();
 
-describe('contract.events [ @E2E ]', function() {
+describe.skip('contract.events [ @E2E ]', function() {
```
Comment (jdevcs) [inline test/e2e.contract.events.js]: are these tests not passing?

**Review by jdevcs — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

**Review by luu-alex — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Pull request merged**